src/App.tsx

```diff
@@ -15,33 +17,125 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+  wordsLength: number,
+) {
+  const visibleGoods = [...goods]
+    .filter(goodItem => goodItem.length >= wordsLength);
+
+  visibleGoods.sort((prev, curr) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return prev.localeCompare(curr);
+      case SortType.LENGTH:
+        return prev.length - curr.length;
+      case SortType.NONE:
+      default:
+        return 0;
+    }
+  });
```

Comment:
This code will work faster because your sorting will not start in case of None or the default, on other hand in your solution it will look at each element and do nothing...
```suggestion
switch (sortType) {
      case SortType.ALPABET:
        return visibleGoods.sort((a, b) => a.localeCompare(b));
      case SortType.LENGTH:
        return visibleGoods.sort((a, b) => a.length - b.length);
      case SortType.NONE:
      default:
        break;
    }
```